init.lua

```diff
@@ -5,6 +5,37 @@ local S = minetest.get_translator(minetest.get_current_modname())
 local F = minetest.formspec_escape
 local C = minetest.colorize
 
+-- Function to split a string by a delimiter and return a table
+local function split_string(str, delimiter)
+    local result = {}
+    for match in string.gmatch(str, "([^"..delimiter.."]+)") do
+        table.insert(result, match)
+    end
+    return result
+end

```

Comment:
I think this can be done via `ItemStack(stack):get_name()`

other than that, minetest has a `string.split()` helper. see https://api.minetest.net/helper-functions/